lxd/instance/drivers/driver_qemu.go

```diff
@@ -4279,21 +4248,24 @@ func (d *qemu) addDriveConfig(qemuDev map[string]any, bootIndexes map[string]int
 		case "cdrom":
 			qemuDev["driver"] = "scsi-cd"
 		}
-	} else if slices.Contains([]string{"nvme", "virtio-blk"}, bus) {
-		qemuDevBus, ok := qemuDev["bus"].(string)
-		if !ok || qemuDevBus == "" {
-			// Try to get a PCI address for hotplugging.
-			pciDeviceName, err := d.getPCIHotplug()
-			if err != nil {
-				return nil, err
-			}
+	} else if slices.Contains([]string{"nvme", "virtio-blk"}, busName) {
+		qemuDev["driver"] = busName
 
-			d.logger.Debug("Using PCI bus device to hotplug drive into", logger.Ctx{"device": driveConf.DevName, "port": pciDeviceName})
-			qemuDev["bus"] = pciDeviceName
-			qemuDev["addr"] = "00.0"
+		// Allocate a device port.
+		devBus, devAddr, multi, err := busAllocate()
+		if err != nil {
+			return nil, fmt.Errorf("Failed allocating bus for disk device %q: %w", driveConf.DevName, err)
 		}
 
-		qemuDev["driver"] = bus
+		d.logger.Debug("Using device port to plug drive into", logger.Ctx{"device": driveConf.DevName, "port": devBus})
+
+		// Populate the qemu device with port info.
+		qemuDev["bus"] = devBus
+		qemuDev["addr"] = devAddr
+
+		if multi {
+			qemuDev["multifunction"] = true
+		}
```

Comment:
```suggestion
		qemuDev["multifunction"] = multi
```

`true` will render to `on` and `false` will simply omit the config line.